### PR TITLE
Remove the round to test allgather with all nodes

### PR DESCRIPTION
**Pull request by workingloong**

### CODE REVIEW

File: dlrover/python/elastic_agent/torch/training.py [hunk 1]
```diff
@@ -575,7 +574,7 @@ def run(self, role: str = DEFAULT_ROLE) -> bool:
             f"{spec.get_entrypoint_name()}"
         )
         success = False
-        for i in range(self._max_check_round):
+        for i in range(2):
```
Comment (merlintang) [inline dlrover/python/elastic_agent/torch/training.py:577]: why change to fixed number?

File: dlrover/python/elastic_agent/torch/training.py [hunk 2]
```diff
@@ -585,11 +584,13 @@ def run(self, role: str = DEFAULT_ROLE) -> bool:
...
             if network_ready:
                 return True
-            elif i == 0 and self._worker_group.group_world_size <= 2:
-                logger.error(
-                    "Fail to check network when there are only 2 nodes."
-                )
-                raise RuntimeError("The node network is breakdown.")
+            else:
+                total_worker_num = len(self._client.get_running_nodes())
+                if total_worker_num <= 2:
```
Comment (merlintang) [inline dlrover/python/elastic_agent/torch/training.py:589]: change this 2 to configurable?

**Review by merlintang — COMMENTED** [review]
leave comment for this pr

File: dlrover/python/elastic_agent/torch/training.py (hunk 1 above)
Comment (workingloong) [reply dlrover/python/elastic_agent/torch/training.py]: Fixed it.

File: dlrover/python/elastic_agent/torch/training.py (hunk 2 above)
Comment (workingloong) [reply dlrover/python/elastic_agent/torch/training.py:591]: Not necessary and I have added annotation to explain the reason.

**Review by hxdtest — APPROVED** [review]

**Pull request merged**